backend/Cache.js

```diff
@@ -0,0 +1,61 @@
+class Cache {
+  constructor(ttl = 3600, maxSize = 100) {
+    this.cache = new Map();
+    this.ttl = ttl * 1000; // Convert TTL to milliseconds.
+    this.maxSize = maxSize; // Maximum number of items in cache
+
+    // Periodically clean up expired entries
+    setInterval(this.cleanup.bind(this), this.ttl);
+  }
+
+  get(key) {
+    const cachedItem = this.cache.get(key);
+    if (!cachedItem) {
+      return null;
+    }
+
+    const [value, expiryTime] = cachedItem;
+
+    if (Date.now() > expiryTime) {
+      // Item has expired
+      this.cache.delete(key);
+      return null;
+    }
+
+    // Update the item's position to implement LRU
+    this.cache.delete(key);
+    this.cache.set(key, [value, expiryTime]);
+
+    return value;
+  }
+
+  set(key, value) {
+    if (this.cache.size >= this.maxSize) {
+      // Evict the least recently used item
+      const firstKey = this.cache.keys().next().value;
```

Comment:
Is this.cache.keys() ordered? Usually when you return from a map the keys are not ordered (or at least not in the way you think it is).

Reply:
yes they are ordered

> The keys() method of [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instances returns a new [map iterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator) object that contains the keys for each element in this map in insertion order.